workers/common/src/main/kotlin/common/env/definition/EnvironmentServiceDefinition.kt

```diff
@@ -40,15 +41,15 @@ open class EnvironmentServiceDefinition(
     val service: InfrastructureService,
 
     /**
-     * A flag to indicate whether the associated [InfrastructureService] should be excluded from the _.netrc_ file.
-     * If this flag is defined, it overrides the corresponding flag from the service.
+     * A set determining the locations where the credentials of the associated [InfrastructureService] should be
+     * used. If it is defined, it overrides the corresponding property from the service.
      */
-    val excludeServiceFromNetrc: Boolean? = null
+    val credentialsTypes: Set<CredentialsType>? = null
```

Comment:
What would it mean if this value is `null`, and what would it mean if it is an empty Set?
Does it mean the same? Is this handled differently?

Could the value be non-nullable and use an emptySet as default?

Reply:
The service definition allows overriding the corresponding property from the infrastructure service if `credentialsType` is not *null*. So, there is a difference between a *null* value and an empty set: *null* means that the value from the infrastructure service is used; otherwise, the value from the definition is used. The comment says "If it is defined, it overrides the corresponding property from the service." Do you think, this needs further clarification?